.github/workflows/pull-request-generate-clients.yml

```diff
@@ -0,0 +1,72 @@
+name: Test Generate API Clients
```

Comment:
I think it would be best to change the actual `generate-clients.yaml` files and add a condition on pull_request to avoid pushing in clients repositories. 
This way we will have to manage only on file for both `main` branch and `pull_requests` generation